crates/sui-json-rpc/src/read_api.rs

```diff
@@ -209,102 +236,230 @@ impl ReadApiServer for ReadApi {
             .collect())
     }
 
-    async fn get_transaction(
+    async fn get_transaction_with_options(
         &self,
         digest: TransactionDigest,
+        opts: Option<SuiTransactionResponseOptions>,
     ) -> RpcResult<SuiTransactionResponse> {
-        let (transaction, effects) = self
-            .state
-            .get_executed_transaction_and_effects(digest)
-            .await
-            .tap_err(|err| debug!(tx_digest=?digest, "Failed to get transaction: {:?}", err))?;
-        let checkpoint = self
-            .state
-            .get_transaction_checkpoint(&digest)
-            .map_err(|e| anyhow!("{e}"))?;
-        let checkpoint_timestamp = checkpoint.as_ref().map(|c| c.summary.timestamp_ms);
+        let opts = opts.unwrap_or_default();
+        let mut response_cache = TempTransactionCache::new(digest);
```

Comment:
to double check, here and in mult_get, `response_cache` is both instantiated, so this is not a cache a following get_txn request, right? What's the benefit of having this then?

Reply:
Good question! The `cache` in the naming is probably confusing since this is not used as a `cache` but a temp data structure to make it easy to work with this bundle of fields, as some fields such as `events` depend on `TransactionEffects` instead of `SuiTransactionEffects`as in `SuiTransactionResponse`. Let me rename the struct type to `IntermediateTransactionResponse` to make this clearer.